packages/contracts-bedrock/src/L1/SystemConfigInterop.sol

```diff
@@ -48,19 +67,63 @@ contract SystemConfigInterop is SystemConfig {
         }
     }
 
-    /// @notice Adds a chain to the interop dependency set. Can only be called by the owner.
+    /// @notice Adds a chain to the interop dependency set. Can only be called by the add dependency role address.
     /// @param _chainId Chain ID of chain to add.
-    function addDependency(uint256 _chainId) external onlyOwner {
+    function addDependency(uint256 _chainId) external {
+        require(
+            msg.sender == Storage.getAddress(ADD_DEPENDENCY_ROLE_HOLDER_SLOT),
+            "SystemConfig: caller is not add dependency role address"
+        );
         OptimismPortal(payable(optimismPortal())).setConfig(
             ConfigType.ADD_DEPENDENCY, StaticConfig.encodeAddDependency(_chainId)
         );
     }
 
-    /// @notice Removes a chain from the interop dependency set. Can only be called by the owner.
+    /// @notice Removes a chain from the interop dependency set. Can only be called by the remove dependency role
+    /// address.
     /// @param _chainId Chain ID of the chain to remove.
-    function removeDependency(uint256 _chainId) external onlyOwner {
+    function removeDependency(uint256 _chainId) external {
+        require(
+            msg.sender == Storage.getAddress(REMOVE_DEPENDENCY_ROLE_HOLDER_SLOT),
+            "SystemConfig: caller is not remove dependency role address"
+        );
         OptimismPortal(payable(optimismPortal())).setConfig(
             ConfigType.REMOVE_DEPENDENCY, StaticConfig.encodeRemoveDependency(_chainId)
         );
     }
+
+    /// @notice Sets the address that can call addDependency. Can only be called by the foundation multisig.
+    /// @param _addDependencyRoleHolder New address authorized to call addDependency
+    function setAddDependencyRoleHolder(address _addDependencyRoleHolder) external {
```

Comment:
What do you think about removing these setters and we can just modify through `initialize` again? That will simplify the management and reduce the ABI we have to maintain

Reply:
this would mean it cannot be modified though, without an upgrade at least, is that okay?

Reply:
seems good to me ✅

Reply:
The dependency manager identity can be modified at initialization but then that identity can add/remove dependencies. The next time that we want to modify the dependency manager, we can have the proxy admin owner call `initialize` again. This keeps the diff small before we commit to a full role based auth system rather than using ownable

Reply:
It is ok that it cannot be modified without an upgrade, we can just upgrade to the same implementation if necessary. In the long term future, we could use something like [AccessControl](https://github.com/OpenZeppelin/openzeppelin-contracts/blob/bcd4beb5e7fd8bd8edf160fbffb5d5b03804efdb/contracts/access/AccessControl.sol) to create specific roles. Doing that now would add complexity, once we identify a 3rd role then we can migrate to using something like `AccessControl`